tests/AMSlib/torch_model.cpp

```diff
@@ -46,6 +46,12 @@ void inference(SurrogateModel<T> &model, AMSResourceType resource)
 int main(int argc, char *argv[])
 {
   using namespace ams;
+  if (argc != 4) {
```

Comment:
Good first check but what if the input is neither "single" or "double" when you `std::strcmp` later on? Add an `else {` failing out as the fall-through path in the string comparisons later on this file

Reply:
Fixed it.